src/subarraynew.jl

```diff
@@ -0,0 +1,36 @@
+typealias NullableSubArray{T,N,P<:NullableArray,IV,LD} SubArray{T,N,P,IV,LD}
+
+@inline function Base.isnull(V::NullableSubArray, I::Real...)
+    @boundscheck checkbounds(V, I...)
+    @inbounds return V.parent.isnull[Base.reindex(V, V.indexes, Base.to_indexes(I...))...]
```

Comment:
@kmsquire I was asked to keep the fixing of the indentation as a separate commit, and I think the fixes related to adding Compat should probably be in a separate commit from the fixes related to JuliaLang/julia#15071, but I'll squash out that commit.  Does that sound reasonable?